script.js

```diff
@@ -0,0 +1,209 @@
+        'Access-Control-Allow-Origin': '*'
+      },
+      body: JSON.stringify({
+        text: input.value
+      })
+    });
+    const data = await result.json();
+    allTasks.push(data);
+    input.value = '';
+    render();
+
+  } catch (error) {
+    showError('Ошибка добавления задачи');
+  }
+};
+
+const deleteTasks = async () => {
+  try {
+    await fetch('http://localhost:8080/tasks', {
+      method: 'DELETE'
+    });
+    allTasks = [];
+    render();
+
+  } catch (error) {
+    showError('Ошибка удаления задач');
+  }
+};
+
+const deleteTask = async (element) => {
+  try {
+    await fetch(`http://localhost:8080/tasks/${element._id}`, {
+      method: 'DELETE',
+    });
+    index = allTasks.indexOf(element);
+    allTasks.slice(index, 1);
+    render();
+  } catch (error) {
+    showError('Ошибка удаления задачи');
+  }
+};
+
+const changeCheckBoxTask = async (element) => {
+  try {
+    const result = await fetch(`http://localhost:8080/tasks/isCheck/${element._id}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        'Access-Control-Allow-Origin': '*'
+      },
+      body: JSON.stringify({
+        isCheck: !element.isCheck
+      })
+    });
+    const data = await result.json();
+
+    allTasks.forEach(item => {
+      if (item._id === element._id) {
+        item.isCheck = data.isCheck;
+      }
+    });
+    render();
+  } catch (error) {
+    showError('Ошибка изменения задачи');
+  }
+};
+
+const changeTextTask = async (element) => {
+  try {
+    const input = document.querySelector('.lists_with_checkBox__input_text');
+    const result = await fetch(`http://localhost:8080/tasks/text/${element._id}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json;charset=utf-8',
+        'Access-Control-Allow-Origin': '*'
+      },
+      body: JSON.stringify({
+        text: input.value
+      })
+    });
+    const data = await result.json();
+
+    allTasks.forEach(item => {
+      if (item._id === element._id) {
+        console.log(data.text);
+        item.text = data.text;
+      }
+    });
+    render();
+  } catch (error) {
+    showError('Ошибка удаления задачи');
+  }
+};
+
+const editTask = (element) => {
+  indexEditableTask = element._id;
+  render(indexEditableTask);
+};
+
+const showError = (error) => {
+  const window = document.createElement('p');
+  const bodyContainer = document.getElementById('forError');
+  const body = document.getElementsByTagName('body')[0];
+
+  window.innerText = error;
+  bodyContainer.className = 'container_error container';
+  window.className = 'error';
+  body.append(window);
+  render();
+};
+
+const render = (indexEditableElement = null) => {
+  const allTasks = document.getElementById('list_with_Tasks');
+  while (allTasks.firstChild) {
+    allTasks.removeChild(allTasks.firstChild);
+  };
+
+  const copyAllTasks = [...allTasks];

```

Comment:
I wonder what exactly are you taking here for allTasks? The array that you set at the beginning of the file or the one here?